tests/wpt/mozilla/tests/mozilla/parser-reentrancy-customelement.html

```diff
@@ -0,0 +1,27 @@
+<html>
+<head>
+<title>Dynamic markup insertion during custom element callbacks does not panic</title>
```

Comment:
Can we get a test with `dom.servoparser.async_html_tokenizer.enabled`? Does the mozilla folder also support adding a `__dir__.ini` in the meta folder? See for example https://github.com/servo/servo/blob/main/tests/wpt/meta/resize-observer/__dir__.ini

Reply:
Good call!

Reply:
I've added the new test with the async parser variant, and it's exposed an issue in the async parser that will need more investigation. I propose merging with the current panicking result and doing the investigation in a separate followup.